### ISSUE TITLE
libxsmm error building with gcc

### ISSUE THREAD

**Issue by lcebaman** [ExtremeFLOW/neko#1615]
Trying to build Neko with libxsmm support (gcc compiler) I got:

```
math/bcknd/xsmm/opr_xsmm.F90:140:47:

  140 |     call col2(du, coef%jacinv, coef%dof%n_dofs)
      |                                               1
Error: ‘n_dofs’ at (1) is not a member of the ‘dofmap_t’ structure; did you mean ‘dof’?
```
Is this known?

**Comment by MartinKarp** [comment]
That seems like that part of the code has not been updated for a while, the correct way nowadays to get the size of the dofmap should be with %size() instead of %n_dofs.

**Comment by lcebaman** [comment]
@MartinKarp Correct. Replacing  `%n_dofs` with `%size()` helps to build the application. I guess that should be changed.

**Comment by njansson** [comment]
> @MartinKarp Correct. Replacing `%n_dofs` with `%size()` helps to build the application. I guess that should be changed.

Great! Yes please consider to submit  a patch if it solved the issue

**Comment by lcebaman** [comment]
Yes, will do
